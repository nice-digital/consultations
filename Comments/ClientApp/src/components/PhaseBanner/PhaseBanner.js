// @flow

import React, { PureComponent } from "react";

type PropsType = {
	phase: string,
	name: string,
}

export class PhaseBanner extends PureComponent<PropsType> {
	render() {
		const {phase, name} = this.props;
		return (
			<div className="container phase-banner-container">
				<div className="grid">
					<div data-g="12">
						<aside className="phase-banner mt--b">
							<span className="phase-banner__tag">
								<span className="tag tag--impact tag--beta">{phase}</span>
							</span>
							<span className="phase-banner__label">
								{name} is in development, which means that some features may not work fully.{" "}
								<a
									href="/get-involved/contact-us"
									rel="noopener noreferrer"
									target="_blank"
								>Report an issue</a>
							</span>
						</aside>
					</div>
				</div>
			</div>
		);
	}
}
