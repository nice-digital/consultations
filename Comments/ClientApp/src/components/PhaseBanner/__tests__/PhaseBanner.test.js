import React from "react";
import { render } from "@testing-library/react";
import { PhaseBanner } from "../PhaseBanner";

test("Matches snapshot", () => {
	const FakeProps = {
		name: "My Project",
		repo: "http://mygitrepo.com",
		phase: "delta",
	};
	const {container} = render(<PhaseBanner {...FakeProps}/>);
	expect(container).toMatchSnapshot();
});
