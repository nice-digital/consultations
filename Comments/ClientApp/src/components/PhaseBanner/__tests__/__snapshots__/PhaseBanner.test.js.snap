// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  PhaseBanner  Matches snapshot 1`] = `
<div
  className="container phase-banner-container"
>
  <div
    className="grid"
  >
    <div
      data-g="12"
    >
      <aside
        className="phase-banner mt--b"
      >
        <span
          className="phase-banner__tag"
        >
          <span
            className="tag tag--impact tag--beta"
          >
            delta
          </span>
        </span>
        <span
          className="phase-banner__label"
        >
          My Project
           is in development, which means that some features may not work fully.
           
          <a
            href="/get-involved/contact-us"
            rel="noopener noreferrer"
            target="_blank"
          >
            Report an issue
          </a>
        </span>
      </aside>
    </div>
  </div>
</div>
`;
